### PR TITLE
Add debug info collection for block device relevant cases

### PR DESCRIPTION
**Pull request by qingwangrh**
Add gstack and strace log collection for cases:
block_hotplug,block_resize,hotplug_unplug_during_io_repeat, multi_disk_random_hotplug,multi_disk_wild_hotplug

ID:1956,1957,1958,1959,1960

### CODE REVIEW

**Comment by zhenyzha** [comment]
These 6 cases all pass on the arm platform

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Comment by XueqiangWei** [comment]
@qingwangrh multi_disk_wild_hotplug.cfg has been updated in another patch, please resolve the conflicts . Thanks.

**Comment by qingwangrh** [comment]
> @qingwangrh multi_disk_wild_hotplug.cfg has been updated in another patch, please resolve the conflicts . Thanks.

updated

**Comment by XueqiangWei** [comment]
(1/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_6
4.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdo
wn.after_unplug.one_pci.q35: STARTED
 (1/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_6
4.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdo
wn.after_unplug.one_pci.q35: PASS (56.67 s)
 (2/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_6
4.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: STARTED
 (2/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_6
4.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: PASS (4282.51 s)
 (3/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_6
4.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: ST
ARTED
 (3/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: PASS (1233.03 s)
 (4/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: STARTED
 (4/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: PASS (686.63 s)
 (5/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: STARTED
 (5/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: PASS (446.32 s)
 (6/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: STARTED
 (6/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: PASS (139.87 s)

File: qemu/tests/block_hotplug.py [hunk 1]
```diff
@@ -177,32 +179,44 @@ def check_disk_size(did, excepted_size):
...
+
+            if sub_test_after_unplug:
+                run_sub_test(sub_test_after_unplug)
+        except Exception as e:
+            pid = vm.get_pid()
+            test.log.debug("Find %s Exception:'%s'.", pid, e)
+            if pid:
+                logdir = test.logdir
+                process.getoutput("gstack %s > %s/gstack.log" % (pid, logdir))
+                process.getoutput("timeout 20 strace -tt -T -v -f -s 32 -p %s -o %s/strace.log" % (pid, logdir))
```
Comment (XueqiangWei) [inline qemu/tests/block_hotplug.py:219]: Recommend a maximum of 79 characters per line, suggest to update it as block_resize.py - line 256. Thanks.

File: qemu/tests/block_hotplug.py [hunk 2]
```diff
@@ -177,32 +179,44 @@ def check_disk_size(did, excepted_size):
...
+                run_sub_test(sub_test_before_unplug)
+
+            plug_block_devices('unplug', unplug_devs)
+            del unplug_devs[:]
+
+            if sub_test_after_unplug:
+                run_sub_test(sub_test_after_unplug)
+        except Exception as e:
+            pid = vm.get_pid()
+            test.log.debug("Find %s Exception:'%s'.", pid, e)
```
Comment (XueqiangWei) [inline qemu/tests/block_hotplug.py:215]: str(e)

File: qemu/tests/block_resize.py [hunk 3]
```diff
@@ -119,130 +119,143 @@ def _check_event_cmd(cmd):
...
+
+            # Verify md5
+            if params.get('md5_test') == 'yes':
+                new_md5 = get_md5_of_file(md5_filename)
+                test.assertTrue(new_md5 == md5, "Unmatched md5: %s" % new_md5)
+
+        session.close()
+    except Exception as e:
+        pid = vm.get_pid()
+        test.log.debug("Find %s Exception:'%s'.", pid, e)
```
Comment (XueqiangWei) [inline qemu/tests/block_resize.py:252]: str(e)

File: qemu/tests/hotplug_unplug_during_io_repeat.py [hunk 4]
```diff
@@ -66,6 +68,18 @@ def _run_iozone_background():
             _check_iozone_status()
             plug.unplug_devs_serial()
             iozone_thread.join(suppress_exception=True)
+    except Exception as e:
+        pid = vm.get_pid()
+        test.log.debug("Find %s Exception:'%s'.", pid, e)
```
Comment (XueqiangWei) [inline qemu/tests/hotplug_unplug_during_io_repeat.py:73]: str(e)

File: qemu/tests/multi_disk_random_hotplug.py [hunk 5]
```diff
@@ -256,95 +258,109 @@ def run_stress_dd():
...
-        test.error("I/O error messages occured in dmesg, "
-                   "check the log for details.")
+        out = session.cmd_output('dmesg')
+        if "I/O error" in out:
+            test.log.warn(out)
+            test.error("I/O error messages occured in dmesg, "
+                       "check the log for details.")
+    except Exception as e:
+        pid = vm.get_pid()
+        test.log.debug("Find %s Exception:'%s'.", pid, e)
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_random_hotplug.py:357]: str(e)

File: qemu/tests/multi_disk_wild_hotplug.py [hunk 6]
```diff
@@ -90,35 +92,48 @@ def _hotunplug_images():
...
-    error_context.context("Check disks in guest.", test.log.info)
-    # re-login in case previous session is expired
-    session = vm.wait_for_login(timeout=int(params.get("relogin_timeout", 60)))
-    new_disks_num = int(session.cmd("lsblk -d -n|wc -l", timeout=300))
-    test.log.info("There are total %d disks after hotplug", new_disks_num)
-    if new_disks_num != disks_num:
-        test.log.warning("Find unmatched disk numbers %d %d", disks_num,
-                         new_disks_num)
+    try:
+        session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_wild_hotplug.py:96]: the following method also work well. All look good to me.
timeout = params.get_numeric("login_timeout", 360)

File: qemu/tests/multi_disk_wild_hotplug.py [hunk 7]
```diff
@@ -90,35 +92,48 @@ def _hotunplug_images():
...
+            time.sleep(unplug_time)
+            error_context.context("Start plug loop:%d" % n, test.log.info)
+            _hotplug_images()
+            time.sleep(plug_time)
+
+        error_context.context("Waiting for %d seconds" % wait_time, test.log.info)
+        time.sleep(wait_time)
+        error_context.context("Check disks in guest.", test.log.info)
+        # re-login in case previous session is expired
+        session = vm.wait_for_login(timeout=int(params.get("relogin_timeout", 60)))
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_wild_hotplug.py:122]: The following method also work well. All look good to me.
timeout = params.get_numeric("relogin_timeout", 60)

File: qemu/tests/multi_disk_wild_hotplug.py [hunk 8]
```diff
@@ -90,35 +92,48 @@ def _hotunplug_images():
...
+        new_disks_num = int(session.cmd("lsblk -d -n|wc -l", timeout=300))
+        test.log.info("There are total %d disks after hotplug", new_disks_num)
+        if new_disks_num != disks_num:
+            test.log.warning("Find unmatched disk numbers %d %d", disks_num,
+                             new_disks_num)
+    except Exception as e:
+        pid = vm.get_pid()
+        test.log.debug("Find %s Exception:'%s'.", pid, e)
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_wild_hotplug.py:130]: str(e)

File: qemu/tests/block_resize.py [hunk 9]
```diff
@@ -119,130 +119,143 @@ def _check_event_cmd(cmd):
...
+
+                if not utils_disk.is_mount(partition, dst=mpoint,
+                                           fstype=fstype, session=session):
+                    res = utils_disk.mount(partition, mpoint,
+                                           fstype=fstype, session=session)
+                    if not res:
+                        test.fail("Mounting data disk was failed! ")
+
+            if params.get('iozone_test') == 'yes':
+                iozone_timeout = float(params.get("iozone_timeout", 1800))
```
Comment (XueqiangWei) [inline qemu/tests/block_resize.py:236]: The following method also work well. All look good to me.
iozone_timeout = params.get_numeric("iozone_timeout", 1800, float)

File: qemu/tests/multi_disk_random_hotplug.py [hunk 10]
```diff
@@ -256,95 +258,109 @@ def run_stress_dd():
...
+                                  % iteration, test.log.info)
+            sub_type = params.get("sub_type_before_plug")
+            if sub_type:
+                error_context.context(context_msg % (sub_type, "before hotplug"),
+                                      test.log.info)
+                utils_test.run_virt_sub_test(test, params, env, sub_type)
+
+            error_context.context("Hotplug the devices", test.log.debug)
+            getattr(plug, hotplug)(timeout=timeout)
+            time.sleep(float(params.get('wait_after_hotplug', 0)))
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_random_hotplug.py:304]: The following method also work well. All look good to me.
time.sleep(params.get_numeric('wait_after_hotplug', 0, float))

File: qemu/tests/multi_disk_random_hotplug.py [hunk 11]
```diff
@@ -256,95 +258,109 @@ def run_stress_dd():
...
+            error_context.context("Unplug and remove the devices", test.log.debug)
+            if stress_cmd:
+                session.cmd(params["stress_stop_cmd"])
+            getattr(plug, unplug)(timeout=timeout, interval=interval_time_unplug)
+            if stress_cmd:
+                session.cmd(params["stress_cont_cmd"])
+            _postprocess_images()
+
+            error_context.context("Verify disks after unplug", test.log.debug)
+            time.sleep(float(params.get('wait_after_unplug', 0)))
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_random_hotplug.py:333]: The following method also work well. All look good to me.
time.sleep(params.get_numeric('wait_after_unplug', 0, float))

**Comment by qingwangrh** [comment]
@zhencliu ,Please help to review, thanks,

**Comment by XueqiangWei** [comment]
Tested with edk2, the results were passed.
(1/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.ext
ra_cdrom_ks.default_install.aio_threads.q35: STARTED
 (1/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.ext
ra_cdrom_ks.default_install.aio_threads.q35: PASS (666.13 s)
 (2/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with
out_delay.q35: STARTED
 (2/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with
out_delay.q35: PASS (450.69 s)
 (3/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_d
ata_disk.fmt_raw.default_partition.q35: STARTED
 (3/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RH
EL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: PASS (142.68 s)
 (4/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: STARTED
 (4/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: PASS (59.85 s)
 (5/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: STARTED
 (5/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: PASS (4318.50 s)
 (6/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: STARTED
 (6/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: PASS (1059.33 s)
 (7/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: STARTED
 (7/7) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: PASS (685.45 s)

**Comment by XueqiangWei** [comment]
tested with seabios, all passed.
 (1/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.
4.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cd
rom_ks.default_install.aio_threads.q35: STARTED
 (1/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.
4.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cd
rom_ks.default_install.aio_threads.q35: PASS (632.82 s)
 (2/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.
4.0.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow
2.with_plug.with_shutdown.after_unplug.one_pci.q35: STARTED
 (2/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.
4.0.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow
2.with_plug.with_shutdown.after_unplug.one_pci.q35: PASS (43.85 s)
 (3/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.
4.0.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q
35: STARTED
 (3/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: PASS (4321.67 s)
 (4/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: STARTED
 (4/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: PASS (99.18 s)
 (5/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: STARTED
 (5/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: PASS (434.68 s)
 (6/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: STARTED
 (6/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: PASS (1044.85 s)
 (7/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: STARTED
 (7/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: PASS (672.07 s)

**Comment by XueqiangWei** [comment]
Tested with win2022, all passed.
(1/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Wi
n2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_
cdrom_ks.default_install.aio_threads.q35: STARTED
 (1/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Wi
n2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_
cdrom_ks.default_install.aio_threads.q35: PASS (1114.49 s)
 (2/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: STARTED
 (2/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.shared_bus.q35: PASS (1695.31 s)
 (3/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: STARTED
(3/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: PASS (3960.23 s)
 (4/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: STARTED
 (4/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: PASS (296.11 s)
 (5/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: STARTED
 (5/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: PASS (4077.21 s)
(6/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: STARTED
 (6/6) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: PASS (144.04 s)

**Comment by XueqiangWei** [comment]
Tested with win2019, all passed.
 (1/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019
.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom
_ks.default_install.aio_threads.q35: STARTED
 (1/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019
.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom
_ks.default_install.aio_threads.q35: PASS (675.86 s)
 (2/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019
.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_typ
e.shared_bus.q35: STARTED   
 (2/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019
.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_typ
e.shared_bus.q35: PASS (1792.13 s)
 (3/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019
.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_typ
e.separated_bus.q35: STARTED
 (3/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.multi_disk_random_hotplug.single_type.separated_bus.q35: PASS (3605.65 s)
 (4/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: STARTED
 (4/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_resize.with_md5.with_data_disk.fmt_raw.default_partition.q35: PASS (262.07 s)
 (5/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: STARTED
(5/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.with_plug.with_shutdown.after_unplug.one_pci.q35: PASS (123.51 s)
 (6/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: STARTED
(6/6) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.hotplug_unplug_during_io_repeat.q35: PASS (4041.86 s)

**Pull request merged**